search-endpoint.md

```diff
@@ -0,0 +1,397 @@
+# Discovery Search Endpoints
+[Back to the list of all defined endpoints](endpoints.md)
+
+## Search Endpoint
+**/api/discover/search**   
+
+This endpoint provides the functionality of the Discovery search screen: http://demo.dspace.org/xmlui/discover.
+
+Provide access to the Discovery search system (SOLR based). This endpoint returns the list of available DSpace object (DSO) types that can be searched:
+* all: Execute a query over all available DSO types.
+* items: Only search the DSpace items.
+* communities: Search within the DSpace community records.
+* collections: Limit the search to DSpace collection records.
+
+Example: TODO
+
+### Search with a given DSO type
+**/api/discover/search/<:dso-type>**
```

Comment:
One of the problems with `/discover/search/objects` is that this maps to the [RestResourceController .findOneInternal](https://github.com/DSpace/DSpace/blob/master/dspace-spring-rest/src/main/java/org/dspace/app/rest/RestResourceController.java#L106) method which does not support Pageable objects (and that's OK since having a Pageable object for a `findOne` method would be weird but it is a problem for this use case).